src/main/kotlin/com/sourcegraph/cody/chat/ContextFilesMessage.kt

```diff
@@ -0,0 +1,73 @@
+package com.sourcegraph.cody.chat
+
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.project.DumbAwareAction
+import com.intellij.openapi.project.Project
+import com.intellij.ui.components.AnActionLink
+import com.intellij.util.ui.JBInsets
+import com.intellij.util.ui.JBUI
+import com.sourcegraph.cody.agent.protocol.ContextFile
+import com.sourcegraph.cody.agent.protocol.ContextMessage
+import com.sourcegraph.cody.agent.protocol.Speaker
+import com.sourcegraph.cody.chat.ChatUIConstants.*
+import com.sourcegraph.cody.ui.AccordionSection
+import java.awt.BorderLayout
+import java.awt.Insets
+import java.util.*
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.stream.Collectors
+import javax.swing.JPanel
+import javax.swing.border.EmptyBorder
+
+class ContextFilesMessage(project: Project, contextMessages: List<ContextMessage>) :
+    PanelWithGradientBorder(ASSISTANT_MESSAGE_GRADIENT_WIDTH, Speaker.ASSISTANT) {
+  init {
+    this.layout = BorderLayout()
+
+    val margin = JBInsets.create(Insets(TEXT_MARGIN, TEXT_MARGIN, TEXT_MARGIN, TEXT_MARGIN))
+    val contextFileNames =
+        contextMessages
+            .stream()
+            .map<ContextFile>(ContextMessage::file)
+            .filter { obj: ContextFile? -> Objects.nonNull(obj) }
+            .map(ContextFile::fileName)
+            .collect(Collectors.toSet())
+
+    val accordionSection = AccordionSection("Read ${contextFileNames.size} files")
+    accordionSection.isOpaque = false
+    accordionSection.border = EmptyBorder(margin)
+    val fileIndex = AtomicInteger(0)
```

Comment:
Suggestion: `AtomicInteger` -> `Int`

(code is synchronous)

Reply:
Or even better: `forEachIndexed`! :smile:

Reply:
ok